contrib/backporting/check-stable

```diff
@@ -156,11 +156,17 @@ generate_commit_list_for_pr () {
     entry_sha=${entry_array[1]}
     entry_sub=${entry_array[@]:2}
     entry_sub_re="^$(sed 's/[.^$*+?()[{\|]/\\&/g' <<< "$entry_sub")$" # adds backslashes for extended regex
-    upstream_commit="$(git log --since="1year" --pretty="%H" --no-merges --extended-regexp --grep "$entry_sub_re" $REMOTE/master)"
-    upstream_id="$(git log -n1 --pretty=format:"%ae%at" $upstream_commit)"
-    if [ "$entry_id" == "${upstream_id}" ]; then
-      echo -e "     |  ${upstream_commit} via $entry_sha (\"$entry_sub\")"
-    else
+    related_commits="$(git log --since="1year" --pretty="%H" --no-merges --extended-regexp --grep "$entry_sub_re" $REMOTE/master)"
+    found_commit=0
+    for upstream_commit in ${related_commits}; do
+      upstream_id="$(git log -n1 --pretty=format:"%ae%at" $upstream_commit)"
+      if [ "$entry_id" == "${upstream_id}" ]; then
+        echo -e "     |  ${upstream_commit} via $entry_sha (\"$entry_sub\")"
+        found_commit=1
+        break
```

Comment:
Should we remove the break here to fail in cases where multiple commits match?

Reply:
:thinking: How can multiple commits match at this point? `$entry_id` should contain the unix timestamp of the commit.

Reply:
If several commits were authored at the same time, with the same author email address, and the same title :grin: 
Probably fine as is :smile:

Reply:
FWIW if we removed the "break", we still would have set `found_commit=1` so it would still just take the first commit rather than exiting out. I guess we could do a counter and ensure we iterate all of the commits and avoid the break, but we're probably solving unrealistic problems at this point ;)